aiida_vasp/workflows/base.py

```diff
@@ -0,0 +1,130 @@
+"""
+Base WorkChain for VASP, Error Handling enriched wrapper around VaspCalculation.
+
+Intended to be reused (launched instead of a VaspCalculation) in all other VASP workflows.
+Any validation and / or error handling that applies to *every* VASP run,
+should be handled on this level, so that every workflow can profit from it.
+Anything related to a subset of use cases must be handled in a subclass.
+"""
+from aiida.work.workchain import WorkChain, ToContext
+from aiida.work import submit
+from aiida.work.db_types import Str
+from aiida.common.extendeddicts import AttributeDict
+from aiida.common.exceptions import NotExistent
+from aiida.orm import Code, CalculationFactory
+
+from aiida_vasp.utils.aiida_utils import get_data_class, builder_interface
+
+
+def get_vasp_proc():
+    vasp_cls = CalculationFactory('vasp.vasp')
+    if builder_interface(vasp_cls):
+        return vasp_cls
+    return vasp_cls.process()
+
+
+class VaspBaseWf(WorkChain):
+    """
+    Error handling enriched wrapper around VaspCalculation.
+
+    Deliberately conserves most of the interface (requied inputs) of the VaspCalculation class.
+
+    This is intended to be used instead of directly submitting a VaspCalculation, so that future features like
+    automatic restarting, error checking etc. can be propagated to higher level workchains automatically by implementing them here.
+
+    Usage::
+
+        from aiida.work import submit
+        basevasp = WorkflowFactory('vasp.base')
+        inputs = basevasp.get_inputs_template  ## AiiDA < 1.0.0a1
+        inputs = basevasp.get_builder() ## AiiDA >= 1.0.0a1
+        ## ... set inputs (documented at aiida.readthedocs.io)
+        submit(basevasp, **inputs)
+    """
```

Comment:
Although there is a more extensive description on how to build workflows in the documentation, I think it would be helpful to mention how to build inputs in as many places as possible. Maybe we can put here a comment to check example/run_wf.py to find out more about the inputs dictionary?